checkov/terraform/parser.py

```diff
@@ -577,6 +625,27 @@ def _clean_parser_types(conf: dict[str, Any]) -> dict[str, Any]:
                 sorted_conf[attribute] = str(values)
         return sorted_conf
 
+    def get_file_key_with_nested_data(self, file, nested_data):
+        if not nested_data:
+            return file
+        nested_str = self.get_file_key_with_nested_data(nested_data.get("file"), nested_data.get('nested_modules_data'))
+        nested_module_name = get_abs_path(nested_str)
+        nested_module_index = nested_data.get('module_index')
+        module_index = get_current_module_index(nested_str)
+        nested_key = nested_str[module_index:]
+        return get_tf_definition_key(file, nested_module_name, nested_module_index, nested_key)
+
+    def get_new_nested_module_key(self, key, file, module_index, nested_data) -> str:
+        if not nested_data:
+            return get_tf_definition_key(key, file, module_index)
+        visited_key_to_add = get_tf_definition_key(key, file, module_index)
+        self.visited_definition_keys.add(visited_key_to_add)
+        nested_key = self.get_new_nested_module_key('', nested_data.get('file'),
+                                                    nested_data.get('module_index'),
+                                                    nested_data.get('nested_modules_data'))
+        new_key = get_tf_definition_key(key, file, module_index, nested_key)
+        return new_key
```

Comment:
```suggestion
        return get_tf_definition_key(key, file, module_index, nested_key)
```
to align it with the method above 😄